### PR TITLE
ci: move scheduled tests into a single workflow

### PR DESCRIPTION
**Pull request by nicoloboschi**

### CODE REVIEW

File: .github/workflows/_run_e2e_tests.yml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+          elif [ "${{ inputs.suite-name == 'langchain' }}" == "true" ]; then
+            commit_ref=$(grep resolved_reference ragstack-e2e-tests/poetry.lock | awk -F'"' '{print $2}' | head -1)
+            echo "commit-url=https://github.com/langchain-ai/langchain/commits/${commit_ref}" >> $GITHUB_OUTPUT
+
+          elif [ "${{ inputs.suite-name == 'llamaindex' }}" == "true" ]; then
+            commit_ref=$(grep resolved_reference ragstack-e2e-tests/poetry.lock | awk -F'"' '{print $2}' | head -1)
+            echo "commit-url=https://github.com/run-llama/llama_index/commits/${commit_ref}" >> $GITHUB_OUTPUT
+
+          elif [ "${{ inputs.suite-name == 'ragstack-latest-release' }}" == "true" ]; then
+            (tox run -c ragstack-e2e-tests -e get-latest-ragstack-ai-version -q) | tail -3 | head -n 1 > /tmp/ragstack-ai-version
```
Comment (jordanrfrazier) [inline .github/workflows/_run_e2e_tests.yml:149]: hm, why does this need the `tail | head`?

File: .github/workflows/ci-e2e-tests.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      astradb: false
+      vector-database-type: "local-cassandra"
+      deploy-to-slack: ${{ needs.preconditions.outputs.is-scheduled == 'true' }}
+      deploy-to-testspace: ${{ needs.preconditions.outputs.is-scheduled == 'true' }}
+      testspace-space: "RAGStack test suite - LangChain dev - DSE"
+
+  langchain-astradev-tests:
+    name: "LangChain dev / AstraDB dev"
+    needs: ["preconditions"]
+    if: ${{ contains(github.event.schedule, '0 */2 * * *') || github.event_name == 'workflow_dispatch' }}
```
Comment (jordanrfrazier) [inline .github/workflows/ci-e2e-tests.yml]: Is this different than
```
${{ needs.preconditions.outputs.is-scheduled == 'true' }}
```

**Review by jordanrfrazier — APPROVED** [review]

File: .github/workflows/ci-e2e-tests.yml (hunk 2 above)
Comment (nicoloboschi) [reply .github/workflows/ci-e2e-tests.yml]: its preparatory for astra prod. we can then add 2 different schedule and trigger jobs based on his own schedule

File: .github/workflows/_run_e2e_tests.yml (hunk 1 above)
Comment (nicoloboschi) [reply .github/workflows/_run_e2e_tests.yml:149]: we want the last 3rd line. it's weird but c'est la vie

**Pull request merged**